src/WiFiCredentials.cpp

```diff
@@ -0,0 +1,194 @@
+#include "WiFiCredentials.h"
+
+#include <Arduino.h>
+#include <LittleFS.h>
+
+#include <esp_log.h>
+
+#include <cstring>
+
+const char* const TAG = "WiFiCredentials";
+
+using namespace OpenShock;
+
+bool WiFiCredentials::Load(std::vector<WiFiCredentials>& credentials) {
+  credentials.clear();
+
+  // Ensure the /wifi/creds directory exists
+  if (!LittleFS.exists("/wifi/creds")) {
+    if (!LittleFS.exists("/wifi")) {
+      if (!LittleFS.mkdir("/wifi")) {
+        ESP_LOGE(TAG, "Failed to create /wifi directory");
+        return false;
+      }
+    }
+    if (!LittleFS.mkdir("/wifi/creds")) {
+      ESP_LOGE(TAG, "Failed to create /wifi/creds directory");
+      return false;
+    }
+    ESP_LOGI(TAG, "No credentials directory found, created one");
+    return true;
+  }
+
+  File credsDir = LittleFS.open("/wifi/creds");
+  if (!credsDir) {
+    ESP_LOGE(TAG, "Failed to open /wifi/creds");
+    return false;
+  }
+
+  while (true) {
+    File file = credsDir.openNextFile();
+    if (!file) {
+      break;
+    }
+
+    WiFiCredentials creds;
+    if (!creds._load(file)) {
+      ESP_LOGE(TAG, "Failed to load credentials from %s", file.name());
+      continue;
+    }
+
+    credentials.push_back(creds);
+  }
+
+  return true;
+}
```

Comment:
Maybe extract the string `"/wifi/creds"` into a constant?

Reply:
@hhvrc

Reply:
Fixed it.